madxtools/tablefs.py

```
@@ -30,85 +30,110 @@
-
-                # Data
-                else:
-                    spLines = tfsLine.split()
-                    assert len(spLines) == len(self.varNames)
-                    for (spLine, vN, vT) in zip(spLines, self.varNames, self.varTypes):
-                        self.Data[vN].append(spLine)
-                        if vT == "%s":
-                            self.Data[vN][-1] = self._stripQuotes(self.Data[vN][-1])
-                    self.nLines += 1
-
-            logger.info("%d lines of data read" % self.nLines)
-
-        if "NAME" in self.Data.keys():
+
+                    # Header/Variable Names
+                    elif tfsLine[0] == "*":
+                        spLines = tfsLine.split()[1:]
+                        for spLine in spLines:
+                            self.varNames.append(spLine)
+                            self.Data[spLine] = []
+
+                    # Header/Variable Type
+                    elif tfsLine[0] == "$":
+                        spLines = tfsLine.split()[1:]
+                        for spLine in spLines:
+                            self.varTypes.append(spLine)
+
+                    # Data
+                    else:
+                        spLines = tfsLine.split()
+                        if len(spLines) != len(self.varNames):
+                            logger.error("Mismatch between data lines and varaible names")
+                            return False
+
+                        for (spLine, vN, vT) in zip(spLines, self.varNames, self.varTypes):
+                            self.Data[vN].append(spLine)
+                            if vT == "%s":
+                                self.Data[vN][-1] = self._stripQuotes(self.Data[vN][-1])
+                        self.nLines += 1
+
+                logger.info("%d lines of data read" % self.nLines)
+
+        except Exception as e:
+            logger.error("Could not read file %s" % self.fileName)
+            logger.error(str(e))
+            return False
```

Isn't it better to just re-raise the error, or to raise some generic TableFSError? A "return False" has no guarantee of being caught

---

Sure. The code was a mix of return status and raised errors. I can make them all raise errors.